torch/_guards.py

```diff
@@ -590,7 +590,7 @@ def __init__(self, fake_mode):
         self.guards_context = GuardsContext()
         self.module_context = ModuleContext()
         self.global_context = GlobalContext()
-        self.fake_mode = fake_mode
+        self.fake_modes = [fake_mode]
```

Comment:
No there should not be multiple modes, there should only ever be two at most, and I'm also not convinced that you need both of them here, it seems to me there should only be one mode in the tracing context (the mutable mode, not the immutable one)

Reply:
(1) I used to set this globally, but you asked for it to be scoped, so now its scoped. LMK which you want
(2) If we compose something else after aot_autograd, you might want to end up pushing another here

I don't really care tho, just lmk how you want it to look, philosophically its all the same to me.